PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift

```diff
@@ -60,22 +59,24 @@ class TagsFilterViewModel: ObservableObject {
     }
 
     func delete(tags: [String]) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsDelete)
-        tracker.track(event: event, [contexts])
+        trackTagsDelete()
         tags.forEach { tag in
             guard let tag: Tag = fetchedTags.filter({ $0.name == tag }).first else { return }
             source.deleteTag(tag: tag)
         }
     }
 
-    func rename(from oldName: String, to newName: String) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsSaveChanges)
-        tracker.track(event: event, [contexts])
-        guard let tag: Tag = fetchedTags.filter({ $0.name == oldName }).first else { return }
+    func rename(from oldName: String?, to newName: String, completion: () -> Void) {
+        guard let oldName else {
+            Log.capture(message: "Unable to rename tag")
```

Comment:
@cyndichin you should call the completion here.

Reply:
I don't want to trigger the code in the completion block, should I return completion(true) or completion(false) in that case then?

Reply:
triggering the code doesn't have an effect from the user perspective, so will add a completion block there as well. thanks!

Reply:
General rule of thumb, I follow is the function here doesn't know what the completion does or doesnt do, so when you return you need to call it. In this case since the completion doesn't take any error parameters I think it should always be called to signify any work this function does is complete.